src/processor/queue-consumer.service.ts

```diff
@@ -164,38 +168,46 @@ export class QueueConsumerService extends WorkerHost implements OnApplicationBoo
     }
   }
 
+  private async setEpochCapacity(totalCapacityUsed: bigint): Promise<void> {
+    const capacityInfo = await this.blockchainService.capacityInfo(this.configService.getProviderId());
+    const currentEpoch = capacityInfo.currentEpoch;
+    const epochCapacityKey = `epochCapacity:${currentEpoch}`;
+    const epochCapacity = BigInt(await this.cacheManager.get(epochCapacityKey) ?? 0);
+    const newEpochCapacity = epochCapacity + totalCapacityUsed;
+    await this.cacheManager.set(epochCapacityKey, newEpochCapacity.toString());
+  }
+
   private async checkCapacity(): Promise<void> {
     const capacityLimit = this.configService.getCapacityLimit();
     const capacity = await this.blockchainService.capacityInfo(this.configService.getProviderId());
     const remainingCapacity = capacity.remainingCapacity;
-    let outOfCapacity = capacity.remainingCapacity <= 0n;
+    const currentEpoch = capacity.currentEpoch;
+    const epochCapacityKey = `epochCapacity:${currentEpoch}`;
+    const epochUsedCapacity = BigInt(await this.cacheManager.get(epochCapacityKey) ?? 0); // Fetch capacity used by the service
+    let outOfCapacity = remainingCapacity <= 0n || epochUsedCapacity >= remainingCapacity;
```

Comment:
if there is no remaining capacity or epochCapacity passed the remaining capacity barrier due the other outside services consuming the capacity too so we pause

Reply:
this is where use of remainingCapacity makes sense